src/move/research/prover-examples/sources/value_holder.move

```diff
@@ -0,0 +1,73 @@
+module prover_examples::value_holder {
+    use std::signer;
+
+    const DEFAULT_VALUE: u8 = 0;
+
+    struct ValueHolder has key {
+        value: u8
+    }
+
+    struct ValueHolderManifest has key {
+        /// The addresses of all accounts that have a ValueHolder.
+        account_addresses: vector<address>,
+        /// The value that all ValueHolders should have.
+        value: u8
+    }
+
+    invariant forall account_address: address where exists<ValueHolder>(
+        account_address
+    ):
+        contains(
+            global<ValueHolderManifest>(@prover_examples).account_addresses,
+            account_address
+        );
```

Comment:
@rahxephon89 the `init_module` and `init_value_holder` functions ensure (as best as I can tell) that if a `ValueHolder` exists under an account, that it was created such that the account address is stored in the `ValueHolderManifest`

So why would this fail?

Reply:
You can write something like:
 ```    
invariant update [suspendable] forall account_address: address where !old(exists<ValueHolder>(
        account_address
    )) && exists<ValueHolder>(
        account_address
    ):
        contains(
            global<ValueHolderManifest>(@prover_examples).account_addresses,
            account_address
        );
``` and add the clause `pragma disable_invariants_in_body;` to the spec of `init_module` and `init_value_holder`.

Reply:
@rahxephon89 thanks for the tip! In 0c762da I actually found that by swapping around the order in which addresses are pushed to the manifests I could eliminate the `suspendable` requirement, but it is useful to know